Hephaestus/Utilities/ConsoleUtility.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+
+namespace Hephaestus.Utilities
+{
+    public static class ConsoleUtility
+    {
+        public static bool AskYesNoQuestion(string question)
+        {
+            bool result;
+            
+            Console.WriteLine($"{question} (Y/N)");
+            ConsoleKey key = Console.ReadKey().Key;
+            
+            // Add some spacing because Console.ReadKey() does not terminate the line
+            Console.WriteLine();
+
+            switch (key)
+            {
+                case ConsoleKey.Y:
+                    result = true;
+                    break;
+                
+                case ConsoleKey.N:
+                    result = false;
+                    break;
+                
+                default:
+                    Console.Error.WriteLine("Incorrect selection. Try again.");
+                    result = AskAgain();
+                    break;
+            }
+
+            return result;
+        }
```

Comment:
This function can be greatly simplified via the use of a do while loop.
1) ask the user the question
2) inside the do while ask for their response
3) check if the response is yes or no with an if statement
4) if it is update the result variable and break out of the while loop
5) return the result
6)if the if statement is not trigger that means there input is invalid and the loop will run again